impeller/renderer/backend/gles/reactor_gles.cc

```diff
@@ -117,18 +123,44 @@ const ProcTableGLES& ReactorGLES::GetProcTable() const {
 }
 
 std::optional<GLuint> ReactorGLES::GetGLHandle(const HandleGLES& handle) const {
+  if (handle.type == HandleType::kFence) {
```

Comment:
The locking and the lookup in the `handles_` map can be extracted out into a separate method in order to reduce the duplication between `ReactorGLES::GetGLHandle` and `GetGLFence`.

Reply:
Done